### PR TITLE
Reference changes and DL page changes.

### LINKED ISSUE #5: Update "Reference" with every new feature added since 2.0

### LINKED ISSUE #19: Add snap store section in download page
See title

### PR DESCRIPTION
**Pull request by phkahler**
Added sections on automatic constraints.
Updated the chord tolerance section to reflect current reality.
Included mention of alternate workplanes for skewed extrusions.
Added Revolve and Helix groups.

### CODE REVIEW

File: ref.pl [hunk 1]
```diff
@@ -687,6 +687,33 @@
     appear on the screen, where the new value can be entered. Press
     enter to commit the change, or Esc to cancel.
 
+<h3>Automatic Constraints</h3>
+
+<p>
+    There is an option in the configuration screens labled "enable
```
Comment (ruevs) [inline ref.pl]: screens = seen

File: ref.pl [hunk 2]
```diff
@@ -687,6 +687,33 @@
...
+    automatic line constraint". If this option is checked and a line
```
Comment (ruevs) [inline ref.pl]: automatic line constraint = automatic line constraint __s__

File: ref.pl [hunk 3]
```diff
@@ -687,6 +687,33 @@
...
+    If an automatic constraint is redundant or would result in the
+    sketch being over-constrained, it will not be created even if
+    it is shown during line placement. Similarly if a distance or
+    diameter constraint is added that would over-constrain the
+    sketch, it will be created as a reference (REF).
+
+<p>
+    Another option in the configuration screen is "edit newly added
+    dimensions". With this option enabled, any time a new distance
+    constraint is added the input box will automatically appear so
```
Comment (ruevs) [inline ref.pl]: With this option enabled, when a new Distance, Length Ratio or Length Difference

File: ref.pl [hunk 4]
```diff
@@ -687,6 +687,33 @@
...
+    a specific value can be entered. Prior to this option the user
```
Comment (ruevs) [inline ref.pl]: With the option off the user...

File: ref.pl [hunk 5]
```diff
@@ -1054,7 +1081,9 @@
     Groups that create a solid (e.g. extrudes or lathes) have a "solid
     model as" option, which is displayed in the page in the text window.
     The group can be merged as union, which adds material to the
-    model, or as difference, which cuts material away.
+    model, or as difference, which cuts material away. A third option is
+    intersection, which preserves only the material that falls within
+    the space of both solids.
```
Comment (ruevs) [inline ref.pl]: the space of both solids. = both solids.

**Review by ruevs — CHANGES_REQUESTED** [review]

**Comment by phkahler** [comment]
@ruevs Thank you for the review! Fixed and added a bit more today.

**Comment by phkahler** [comment]
We can merge these without appearing on the offical site?

**Comment by jwesthues** [comment]
Correct--the live site doesn't update until someone sshs in and types `git pull` there. If you send me an ssh public key I can give you that access, or just let me know when you're ready and I can do it. Until then, feel free to merge these with no risk of any effect on the live site.

**Pull request merged**